.github/workflows/test.yml

```diff
@@ -0,0 +1,30 @@
+name: Run tests
+on: [push]
+env:
+  NODE_VERSION: 20.x
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+
+    permissions:
+      id-token: write
+      contents: read
+
+    steps:
+      - name: checkout commit
+        uses: actions/checkout@v4
+
+      - name: Use Node.js 20.x
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+          cache: npm
+
+      - name: install node deps
+        run: npm ci
+
+      - name: run lint
+        run: npm run lint
+
+      - name: run tests
+        run: npm test
```

Comment:
does this cause lint to be run twice since you have `pretest` to run `lint` in the package.json?

Reply:
Yes, good eye! Updated